api/tests/utils/test_date_utils.py

```diff
@@ -0,0 +1,35 @@
+from utils.date_utils import valid_iso_date
+
+
+def test_valid_iso_date_valid_format():
+    """Test valid_iso_date function with valids ISO 8601 date formats."""
+    # Validators are not required to check for None or empty strings
+    assert valid_iso_date("")
+    assert valid_iso_date("   ")
+    assert valid_iso_date(None)
+
+    assert valid_iso_date("2021-01-01T00:00:00")
+    assert valid_iso_date("2021-01-01T00:00:00Z")
+    assert valid_iso_date("2021-01-01T00:00:00+00:00")
+    assert valid_iso_date("2021-01-01T00:00:00-00:00")
+    assert valid_iso_date("2021-01-01T00:00:00+01:00")
+    assert valid_iso_date("2021-01-01T00:00:00-01:00")
+    assert valid_iso_date("2021-01-01T00:00:00.000Z")
+    assert valid_iso_date("2021-01-01T00:00:00.000+00:00")
+    assert valid_iso_date("2021-01-01T00:00:00.000-00:00")
+    assert valid_iso_date("2021-01-01T00:00:00.000+01:00")
+    assert valid_iso_date("2021-01-01T00:00:00.000-01:00")
+    assert valid_iso_date("2021-01-01T00:00:00.000000Z")
+    assert valid_iso_date("2021-01-01T00:00:00.000000+00:00")
+    assert valid_iso_date("2021-01-01T00:00:00.000000-00:00")
+    assert valid_iso_date("2021-01-01T00:00:00.000000+01:00")
+    assert valid_iso_date("2021-01-01T00:00:00.000000-01:00")
+    assert valid_iso_date("2021-01-01T00:00:00.000000000Z")
+    assert valid_iso_date("2021-01-01T00:00:00.000000000+00:00")
+    assert valid_iso_date("2021-01-01T00:00:00.000000000-00:00")
+
+
+def test_invalid_iso_date_valid_format():
+    """Test valid_iso_date function with invalids ISO 8601 date formats."""
+    assert not valid_iso_date("2021-01-01")
```

Comment:
I thought I saw that just a date was also in conformity with ISO 8601

Reply:
It's a valid ISO 8601 under _reduced precission_ .

Reply:
>For reduced [precision](https://en.wikipedia.org/wiki/Precision_and_accuracy),[[18]](https://en.wikipedia.org/wiki/ISO_8601#cite_note-AccuracyVsPrecision-18) any number of values may be dropped from any of the date and time representations, but in the order from the least to the most significant. For example, "2004-05" is a valid ISO 8601 date, which indicates May (the fifth month) 2004. This format will never represent the 5th day of an unspecified month in 2004, nor will it represent a time-span extending from 2004 into 2005.

If we allow _reduced precision_ we will need to allow dates like 2024-05, 2025, etc. At this point, it's easier to be stricter than trying to satisfy multiple combinations of dates. It's not unreasonable to ask API consumers to provide a valid date-time format.